src/server/json_family.cc

```diff
@@ -326,8 +420,80 @@ OpResult<string> OpDoubleArithmetic(const OpArgs& op_args, string_view key, stri
   return output.as_string();
 }
 
+OpResult<long> OpDel(const OpArgs& op_args, string_view key, string_view path) {
+  long total_deletions = 0;
+  if (path.empty()) {
+    auto& db_slice = op_args.shard->db_slice();
+    auto fres = db_slice.FindExt(op_args.db_ind, key);
+    if (IsValid(fres.first)) {
+      total_deletions += long(db_slice.Del(op_args.db_ind, fres.first));
+    }
+
+    return total_deletions;
+  }
+
+  OpResult<json> result = GetJson(op_args, key);
+  if (!result) {
+    return total_deletions;
+  }
+
+  vector<string> deletion_items;
+  auto cb = [&](const string& path, json& val) { deletion_items.emplace_back(path); };
+
+  json j = result.value();
+  error_code ec = JsonReplace(j, path, cb);
+  if (ec) {
+    VLOG(1) << "Failed to evaulate expression on json with error: " << ec.message();
+    return total_deletions;
```

Comment:
are you saying that even for the invalid path we do not return to a client any errors?

Reply:
My inital implementation returned an error if a key does not exist or a path is invalid.
After, I checked the function return code in ElastiCache documentation and tried to get an error on my local Redis server,
I've found that the function always returns an integer that describes how many keys have been deleted and does not return any errors.